### ISSUE TITLE
CucumberReport ends in NullPointerException since 3.10.2

### ISSUE THREAD

**Issue by ldaume** [jenkinsci/cucumber-reports-plugin#182]
After the update from 3.10.1 to 3.10.2 the acceptance test report failed with:

```
ERROR: Build step failed with exception
java.lang.NullPointerException: Name is null
	at java.lang.Enum.valueOf(Enum.java:236)
	at net.masterthought.cucumber.sorting.SortingMethod.valueOf(SortingMethod.java:8)
	at net.masterthought.jenkins.CucumberReportPublisher.generateReport(CucumberReportPublisher.java:271)
	at net.masterthought.jenkins.CucumberReportPublisher.perform(CucumberReportPublisher.java:219)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:81)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:735)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:676)
	at hudson.model.Build$BuildExecution.post2(Build.java:186)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:621)
	at hudson.model.Run.execute(Run.java:1760)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:405)
Build step Cucumber reports marked build as failure
Finished: FAILURE
```

After the downgrade to 3.10.1 everything works fine ;)

```
[CucumberReport] Found 3 pending steps, while expected not more than 0
[CucumberReport] Build status is left unchanged
Finished: SUCCESS
```

**Comment by msymons** [comment]
Starting with plugin v3.10.2, all my reports have also been failing with similar exceptions on Jenkins v2.46.2 and v2.60.3 (identical up to "at hudson.tasks.." and differing thereafter).  

This seems to be a regression of #176.

A couple observations:

1.  Only "classic jobs" throw the exception...  pipeline jobs seem OK.
2.  There is an easy work-around.  Simply edit the job and toggle the Cucumber reports sorting order (eg, switch from alphabetical to natural and back to alphabetical) and then save the job config.  Cucumber reports then seems to work fine.

**Comment by damianszczepanik** [comment]
Duplicate of #176

**Comment by morganchristiansson** [comment]
#176 was closed and fix released in 3.10.2.

This issue is reporting that it's still occuring in 3.10.2. Can confirm this is happening.

**Comment by damianszczepanik** [comment]
Yes, this may happen. 3.10.3 should resolve problems forever!
